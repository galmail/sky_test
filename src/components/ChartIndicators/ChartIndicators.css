.chart-indicators {
    float: left;
    border: 1px solid gray;
    border-radius: 5px;
    padding: 10px;
}

.chart-indicators > div {
    margin-bottom: 5px;
}

.chart-indicators > div:hover {
    background-color: yellow;
}