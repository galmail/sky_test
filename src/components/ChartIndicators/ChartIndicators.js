import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './ChartIndicators.css';

class ChartIndicators extends Component {
  render() {
    const minTemp = this.props.min != null ? this.props.min : '-----';
    const maxTemp = this.props.max != null ? this.props.max : '-----';
    const avgTemp = this.props.avg != null ? this.props.avg : '-----';
    return (
      <div className="chart-indicators">
        <div className="min-temperature" onMouseOver={() => this.props.onHover('min')} onMouseOut={() => this.props.onHover()}>Min Temp: {minTemp}</div>
        <div className="max-temperature" onMouseOver={() => this.props.onHover('max')} onMouseOut={() => this.props.onHover()}>Max Temp: {maxTemp}</div>
        <div className="avg-temperature" onMouseOver={() => this.props.onHover('avg')} onMouseOut={() => this.props.onHover()}>Avg Temp: {avgTemp}</div>
      </div>
    );
  }
}

ChartIndicators.propTypes = {
  min: PropTypes.number,
  max: PropTypes.number,
  avg: PropTypes.number,
  onHover: PropTypes.func,
};

ChartIndicators.defaultProps = {
  onHover: () => {}
};

export default ChartIndicators;
