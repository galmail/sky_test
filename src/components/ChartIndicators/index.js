import ChartIndicators from './ChartIndicators';
export default ChartIndicators;