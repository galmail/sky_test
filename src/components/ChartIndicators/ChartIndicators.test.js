import React from 'react';
import { shallow } from 'enzyme';

import ChartIndicators from './ChartIndicators';

describe('Test ChartIndicators Component', () => {

  it('should render without crashing', () => {
    const wrapper = shallow(<ChartIndicators />);
    expect(wrapper).toMatchSnapshot();
  });

});
