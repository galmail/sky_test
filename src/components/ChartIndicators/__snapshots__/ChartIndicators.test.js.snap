exports[`Test ChartIndicators Component should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartIndicators
    onHover={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="min-temperature"
          onMouseOut={[Function]}
          onMouseOver={[Function]}>
          Min Temp: 
          -----
</div>,
        <div
          className="max-temperature"
          onMouseOut={[Function]}
          onMouseOver={[Function]}>
          Max Temp: 
          -----
</div>,
        <div
          className="avg-temperature"
          onMouseOut={[Function]}
          onMouseOver={[Function]}>
          Avg Temp: 
          -----
</div>,
      ],
      "className": "chart-indicators",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Min Temp: ",
            "-----",
          ],
          "className": "min-temperature",
          "onMouseOut": [Function],
          "onMouseOver": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Min Temp: ",
          "-----",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Max Temp: ",
            "-----",
          ],
          "className": "max-temperature",
          "onMouseOut": [Function],
          "onMouseOver": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Max Temp: ",
          "-----",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Avg Temp: ",
            "-----",
          ],
          "className": "avg-temperature",
          "onMouseOut": [Function],
          "onMouseOver": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Avg Temp: ",
          "-----",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="min-temperature"
            onMouseOut={[Function]}
            onMouseOver={[Function]}>
            Min Temp: 
            -----
</div>,
          <div
            className="max-temperature"
            onMouseOut={[Function]}
            onMouseOver={[Function]}>
            Max Temp: 
            -----
</div>,
          <div
            className="avg-temperature"
            onMouseOut={[Function]}
            onMouseOver={[Function]}>
            Avg Temp: 
            -----
</div>,
        ],
        "className": "chart-indicators",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Min Temp: ",
              "-----",
            ],
            "className": "min-temperature",
            "onMouseOut": [Function],
            "onMouseOver": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Min Temp: ",
            "-----",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Max Temp: ",
              "-----",
            ],
            "className": "max-temperature",
            "onMouseOut": [Function],
            "onMouseOver": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Max Temp: ",
            "-----",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Avg Temp: ",
              "-----",
            ],
            "className": "avg-temperature",
            "onMouseOut": [Function],
            "onMouseOver": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Avg Temp: ",
            "-----",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
